src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,170 @@
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db
+        random-id                   :random-id} _]
+    ;; this event calls itself at regular intervals
+    ;; TODO: this was previously using setInterval explicitly, with
+    ;; dispatch-later it is using it implicitly. setInterval is
+    ;; problematic for such long period of time and will cause a warning
+    ;; for Android in latest versions of react-native
+    {::request-discoveries {:current-public-key current-public-key
+                            :web3               web3
+                            :contacts           contacts
+                            :message-id         random-id}
+     :dispatch-later [{:ms       (* request-discoveries-interval-s 1000)
+                       :dispatch [:request-discoveries]}]}))
+
+(handlers/register-handler-fx
+  :discoveries-send-portions
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db}
+       [_ to]]
+    {::send-portions {:current-public-key current-public-key
+                      :web3 web3
+                      :contacts contacts
+                      :to to}}))
+
+(handlers/register-handler-fx
+  :discoveries-request-received
+  (fn [{{:keys [current-public-key web3]
+         :contacts/keys [contacts]} :db}
+       [_ {:keys [from]}]]
+    {::send-portions {:current-public-key current-public-key
+                      :web3 web3
+                      :contacts contacts
+                      :to from}}))
+
+(handlers/register-handler-fx
+  :discoveries-response-received
+  [(re-frame/inject-cofx :now)]
+  (fn [{{:keys [discoveries]
+         :contacts/keys [contacts]} :db
+        now                         :now}
+       [_ {:keys [payload from]}]]
+    (when (get contacts from)
+      (when-let [data (:data payload)]
+        (doseq [{:keys [message-id] :as discover} data]
+          (when (and (not (discoveries/exists? message-id))
+                     (not (get discoveries message-id)))
+            (let [discover (assoc discover :created-at now)]
+
+              {:dispatch [:add-discover discover]})))))))
+
+(handlers/register-handler-fx
+  :status-received
+  [(re-frame/inject-cofx :now)]
+  (fn [{{:keys [discoveries]} :db
+        now                   :now}
+       [_ {:keys [from payload]}]]
+    (when (and (not (discoveries/exists? (:message-id payload)))
+               (not (get discoveries (:message-id payload))))
+      (let [{:keys [message-id status hashtags profile]} payload
+            {:keys [name profile-image]} profile
+            discover {:message-id   message-id
+                      :name         name
+                      :photo-path   profile-image
+                      :status       status
+                      :whisper-id   from
+                      :tags         (map #(hash-map :name %) hashtags)
+                      :created-at   now}]
+        {:dispatch [:add-discover discover]}))))
```

Comment:
Why do we need to dispatch `:add-discover` when we can extract its handler into named function and call it here directly to produce effects map ?